Code/Tools/FBuild/FBuildCore/Graph/ObjectNode.cpp

```diff
@@ -1728,7 +1740,36 @@ bool ObjectNode::BuildPreprocessedOutput( const Args & fullArgs, Job * job, bool
     // take a copy of the output because ReadAllData uses huge buffers to avoid re-sizing
     TransferPreprocessedData( ch.GetOut().Get(), ch.GetOutSize(), job );
 
-    return true;
+	return true;
+}
+
+// PreProcessing for SimpleDistribution is just loading the source file
+//------------------------------------------------------------------------------
+bool ObjectNode::LoadStaticSourceFileForDistribution(const Args & fullArgs, Job * job, bool useDeoptimization) const
+{
+	const bool useDedicatedPreprocessor = (GetDedicatedPreprocessor() != nullptr);
+	EmitCompilationMessage(fullArgs, useDeoptimization, false, false, useDedicatedPreprocessor);
```

Comment:
I think the dedicated preprocessor can never be used, since we're skipping preprocessing. I think we can assume false here, or better still, emit an error during bff parsing if the option is set. What do you think?